core/util/remote.js

```diff
@@ -0,0 +1,61 @@
+const fs = require('./fs');
+
+/**
+ * Extract jsonReport from the jsonpReport
+ *
+ * @param {String} jsonpReport - jsonpReport `report(${jsonReport});`
+ * @return {Object} an object of jsonReport
+ */
+function extractReport (jsonpReport) {
+  const jsonReport = jsonpReport.substring(7, jsonpReport.length - 2);
+  return JSON.parse(jsonReport);
+}
+
+/**
+ * Helper function to modify the test status of the JSONP report based on the approved file name.
+ *
+ * @param {String} originalJsonpReport - jsonpReport `report(${jsonReport});`
+ * @param {String} approvedFileName - the name of the screenshot that is approved
+ * @return {String} jsonpReport - modified jsonpReport
+ */
+function modifyJsonpReportHelper (originalJsonpReport, approvedFileName) {
+  const report = extractReport(originalJsonpReport);
+  report.tests.forEach(test => {
+    if (test.pair.fileName === approvedFileName) {
+      test.status = 'pass';
+      delete test.pair.diffImage;
```

Comment:
Can `test.pair` be undefined or falsy here ?

Reply:
`tests` has the following format. This function is triggered when a user clicks on the `Approve` button, which means there is a valid test. So I don't see any scenarios that could have no `test.pair`.

```
  "tests": [
    {
      "pair": {
        "reference": "../bitmaps_reference/backstop_default_BackstopJS_Homepage_0_document_0_phone.png",
        "test": "../bitmaps_test/20200219-232153/backstop_default_BackstopJS_Homepage_0_document_0_phone.png",
        "fileName": "backstop_default_BackstopJS_Homepage_0_document_0_phone.png",
        ...
      },
      "status": "pass"
    },
   ...
]
```